install.ts

```diff
@@ -0,0 +1,185 @@
+const WIN_ZIP: string = "deno_win_x64.zip";
+
+function panic(err: Error): void {
+  if (err) console.error("[deno-install error]", err.stack);
+  console.error("[deno-install error]", "update failed");
+  exit(1);
+}
+
+function pinup(...args: any): void {
+  console.log("[deno-install info]", ...args);
+}
+
+async function follow(url: string): Promise<any> {
+  let located: boolean = false;
+  let res: any; // TODO: annotate deno Response
+  let count: number = 0;
+  while (!located) {
+    if (++count === 4) throw Error(`unable to fetch from ${url}`);
+    res = await fetch(url);
+    if (res.status >= 300 && res.status < 400)
+      url = res.headers.get("Location");
+    else if (res.status === 200) located = true;
+  }
+  return res;
+}
+
+async function release_url(tag?: string): Promise<{ [key: string]: string }> {
+  const url: string = tag ? `${TAG_RELEASE_URL}/${tag}` : LATEST_RELEASE_URL;
+  let filename: string;
+  switch (platform.os) {
+    case "linux":
+      filename = LINUX_GZIP;
+      break;
+    case "mac":
+      filename = OSX_GZIP;
+      break;
+    case "win":
+      filename = WIN_ZIP;
+      break;
+    default:
+      throw Error(`unsupported OS ${platform.os}`);
+  }
+  const res: any = await follow(url); // TODO: annotate deno Response
+  const link: string = (await res.text())
+    .split(/\r?\n/)
+    .find((line: string) => line.includes("href") && line.includes(filename));
+  if (!link) panic(Error(`can't find ${filename} @ ${url}`));
+  const match: string = link.replace(/^.*href=(?:"|')([^"']*)(?:"|').*$/, "$1");
+  if (!/^\/denoland/.test(match))
+    panic(Error(`can't find ${filename} @ ${url}`));
+  return {
+    url: `https://github.com${match}`,
+    tag: match.replace(/^.*(v\d+\.\d+\.\d+).*$/, "$1")
+  };
+}
+
+async function temp_download(
+  temp_dir: string,
+  url: string,
+  suffix: string
+): Promise<string> {
+  const res: any = await follow(url); // TODO: annotate deno Response
+  const temp_file: string = join(temp_dir, `${Date.now()}.${suffix}`);
+  await writeFile(temp_file, new Uint8Array(await res.arrayBuffer()));
+  return temp_file;
+}
+
+async function unpack_bin(archive: string): Promise<void> {
+  await mkdirp(DENO_BIN_DIR);
+  let args: string[];
+  if (WIN32) {
+    await rename(DENO_BIN, OLD_DENO_BIN);
+    args = [
+      "powershell.exe",
+      "-Command",
+      `Expand-Archive "${archive}" -DestinationPath "${DENO_BIN_DIR}"`
+    ];
+  } else {
+    args = ["gunzip", "-d", archive];
+  }
+  const child: Process = run({ args });
+  const child_status: ProcessStatus = await child.status();
+  if (!child_status.success)
+    panic(Error(`(g)unzip failed with code ${child_status.code}`));
+  if (!WIN32) {
+    const gunzipd: string = archive.replace(/\.gz$/, "");
+    if (platform.os === "linux") await rename(DENO_BIN, OLD_DENO_BIN);
+    await copyFile(gunzipd, DENO_BIN);
+  }
+  child.close();
+}
+
+async function make_handy(): Promise<void> {
```

Comment:
This `make_handy` function is not consistent with the other installers (which, too, are not consistent). The Python installer prints a message that the user should add `deno` to the `PATH` manually. The PowerShell installer directly adds `deno` to the `PATH` (since it's kind of easy to do with PowerShell).

I don't think `chmod` is needed on Windows. And I don't think `symlink`ing to System32 is common practice for installers on Windows either.

@chiefbiiko What's your opinion on how to proceed with this?

Reply:
IMHO ideally the installers should not require any manual steps
I found cross-platform path manipulation a little hard to implement, and ended up `symlink`ing `/usr/local/bin/deno` on linux and mac, which works just fine.
For windows path manipulation with powershell seems best to me.
Reflected this in the updated `make_handy` function.